### PR TITLE
Inherit log level in Build pods from OpenShift

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@bparees @soltysh PTAL

File: pkg/build/controller/strategy/docker.go [hunk 1]
```diff
@@ -35,6 +36,7 @@ func (bs *DockerBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod,
 					Image: bs.Image,
 					Env: []kapi.EnvVar{
 						{Name: "BUILD", Value: string(data)},
+						{Name: "LOGLEVEL", Value: string(cmdutil.GetLogLevel())},
```
Comment (smarterclayton) [inline pkg/build/controller/strategy/docker.go]: BUILD_LOGLEVEL

**Comment by bparees** [comment]
lgtm, other than @smarterclayton 's comment.

**Comment by soltysh** [comment]
The idea behind this PR is OK for OpenShift development (mostly builds related), but what about the final user actually building an image inside OpenShift, how he will be able to influence that loglevel? What if in production that loglevel will be set to 0 and the user will try to debug why his build is failing and there will be no log? Previously hardcoded 5 gave that user full output of a build and this does not. Unless I'm missing something here.

**Comment by bparees** [comment]
@soltysh wouldn't they set the LOGLEVEL (BUILD_LOGLEVEL) env variable on the sti buildstrategy?

**Comment by mfojtik** [comment]
@soltysh I think the current log level is pretty noisy and it is very easy to miss some important warning or error. It is possible for STI to show **only** the assemble script output? If is, what is the log level for that?

**Comment by mfojtik** [comment]
@soltysh || @bparees PTAL

Changes:

* The strategy.Env is now *merged** instead of appended to the Build pod, which avoid duplicates (and users can override the default values we set)
* Renamed LOGLEVEL to BUILD_LOGLEVEL

**Comment by soltysh** [comment]
@bparees now when it's merging env vars it is :smile: thx @mfojtik LGTM.
[merge]

**Comment by mfojtik** [comment]
@bparees is his related to the race?

```
E0325 09:47:47.387749   10426 image_change_controller.go:65] unable to find tagged image: no image recorded for default/ruby-20-centos7-buildcli:invalid
```

**Comment by mfojtik** [comment]
[merge] again

**Comment by bparees** [comment]
@mfojtik all i can tell you is it's not the race i was looking at.

**Comment by mfojtik** [comment]
@bparees then I think we have new set of races we need to track :-(

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
last [merge]

**Pull request merged**